### PR TITLE
PSQLODBC Datatype Test for sys.binary & sys.varbinary

### PR DESCRIPTION
**Pull request by ColinKYuen**
### Description

- Adds ODBC Datatype Test for `sys.binary` & `sys.varbinary`
- Added edge cases for timezone helper in `sys.datetimeoffset`
- Removed outdated comments in `sys.datetimeoutset`
- Added driver check to `sys.real`'s teardown

Task: BABELFISH-568 & BABELFISH-572
Signed-off-by: Colin Yuen [yuenhcol@amazon.com](mailto:yuenhcol@amazon.com)

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: test/odbc/psqlodbc/datetimeoffset.cpp [hunk 1]
```diff
@@ -46,10 +46,18 @@ string getTimeZone(const string& date_time) {
...
+      ret.insert(0, "+00");
+      break;
+    case 1:
+      ret.insert(0, "+0");
+      break;
+    case 2:
+      ret.insert(1, "0");
   }
-  return " +0" + ret + ":00";
+  return " " + ret + ":00";
```
Comment (faviansamatha) [inline test/odbc/psqlodbc/datetimeoffset.cpp:60]: A general comment is If we're modifying other files that aren't related to sys.binary or sys.varbinary, could we also add it to he PR description? (and the final commit message as well when squash merging). This is the same as real.cpp

File: test/odbc/psqlodbc/psqlodbc_tests_common.cpp [hunk 2]
```diff
@@ -1,5 +1,42 @@
 #include "psqlodbc_tests_common.h"
 
+std::string GetHexRepresentation(std::string inserted_int, size_t table_size) {
```
Comment (faviansamatha) [inline test/odbc/psqlodbc/psqlodbc_tests_common.cpp:3]: I think we can make these parameters read only (const and &)

File: test/odbc/psqlodbc/psqlodbc_tests_common.h [hunk 3]
```diff
@@ -210,9 +220,11 @@ void testTableCreationFailure(ServerType serverType, const string &tableName, co
  * @param expectedInsertedValues The values expected from the table when selecting all from it.
  * @param pkStartingValue Optional. The primary key value the object starts incrementing at. The default value is 0.
  * @param caseInsensitive Optional. String comparision for data and expected can be case-insensitive. The default value is false.
+ * @param numericInsert Optional. Allow inserts without quotes. e.g. inserting using integers. The default value is false.
 */
 void testInsertionSuccess(ServerType serverType, const string &tableName, const string &orderByColumnName, 
-  const vector<string> &insertedValues, const vector<string> &expectedInsertedValues, int pkStartingValue = 0, bool caseInsensitive = false);
+  const vector<string> &insertedValues, const vector<string> &expectedInsertedValues, int pkStartingValue = 0,
+  bool caseInsensitive = false, bool numericInsert = false);
```
Comment (faviansamatha) [inline test/odbc/psqlodbc/psqlodbc_tests_common.h:227]: I'm wondering if `numericInsert` should be generalized to something like `useQuotedInsert`, `useSingleQuotes` or something like that. In the case that I don't want quotes but inserting something like `CAST('hi' as varbinary())` would not need a single quote around the cast and I would have to set numericInsert to true even though it isn't a numeric insert.

File: test/odbc/psqlodbc/datetimeoffset.cpp (hunk 1 above)
Comment (ColinKYuen) [reply test/odbc/psqlodbc/datetimeoffset.cpp:60]: Done.

File: test/odbc/psqlodbc/psqlodbc_tests_common.cpp (hunk 2 above)
Comment (ColinKYuen) [reply test/odbc/psqlodbc/psqlodbc_tests_common.cpp:3]: Causes issues with the results.

File: test/odbc/psqlodbc/psqlodbc_tests_common.h (hunk 3 above)
Comment (faviansamatha) [reply test/odbc/psqlodbc/psqlodbc_tests_common.h:227]: nvm this comment let’s keep it consistent with the other ones.

**Review by faviansamatha — APPROVED** [review]

**Review by leszekamz — APPROVED** [review]

**Pull request merged**